frontend/src/metabase/parameters/utils/dashboards.ts

```diff
@@ -4,8 +4,51 @@ import { setIn } from "icepick";
 import Question from "metabase-lib/lib/Question";
 import { getParameterTargetField } from "metabase/parameters/utils/targets";
 import { slugify } from "metabase/lib/formatting";
+import {
+  UiParameter,
+  FieldFilterUiParameter,
+  ParameterWithTarget,
+} from "metabase/parameters/types";
+import {
+  Parameter,
+  ParameterMappingOptions,
+  ParameterTarget,
+} from "metabase-types/types/Parameter";
+import {
+  Dashboard,
+  DashboardParameterMapping,
+  DashboardOrderedCard,
+} from "metabase-types/api";
+import { SavedCard } from "metabase-types/types/Card";
+import Metadata from "metabase-lib/lib/metadata/Metadata";
+import Field from "metabase-lib/lib/metadata/Field";
+
+type ExtendedMapping = {
```

Comment:
The "extended" types I've defined here will go away once I refactor and simplify some of the util functions in this file.